### ISSUE TITLE
SAF Support

### ISSUE THREAD

**Issue by crutchcorn** [itinance/react-native-fs#926]
Does the project have any ability to support SAF read/writes? I've tried using a SAF Directory Picker I wrote:

https://github.com/oceanbit-dev/react-native-directory-picker/blob/master/android/src/main/java/com/rndirectorypicker/DirectoryPickerModule.java

But whenever utilizing the `writeFile` or `readFile`, I seem to always get errors since the paths cannot be converted from SAF to `/typical/paths/like/this`. SAF is required as-of Android R, and once React Native apps hit API 30 support, apps that aren't using SAF will likely break (such as mine).

Are there known bugs regarding SAF? What's the future of this package look like?

**Comment by crutchcorn** [comment]
I have also confirmed that even without using a directory picker, the following paths do not work after targeting past API 29:

- `ExternalDirectoryPath`
- `ExternalStorageDirectoryPath`

When I try to run either of these interactions with the following code:

```
  const path = RNFS.ExternalStorageDirectoryPath  + '/images/test.txt';

  const writeToFile = () => {
    RNFS.mkdir(RNFS.ExternalStorageDirectoryPath  + '/images')
      .then(() => RNFS.writeFile(path, 'Lorem ipsum dolor sit amet', 'utf8'))
      .then((success) => {
        console.log('FILE WRITTEN!');
      })
      .catch((err) => {
        console.error(err.message);
      });
  };
```

Which worked on the same emulated Android R device when built using API 28

I get the following error:

```
[Wed Sep 09 2020 20:04:40.864]  ERROR    Directory could not be created
```

![image](https://user-images.githubusercontent.com/9100169/92677076-c843d300-f2d7-11ea-8abf-ac59213006ec.png)

However, interestingly, the following DOES work:

- `DownloadDirectoryPath`

**Comment by tolgaerdonmez** [comment]
@crutchcorn So does it only work on, `DownloadDirectoryPath`, I mean can just use `DownloadDirectoryPath` to write and read files ?

**Comment by crutchcorn** [comment]
@tolgaerdonmez yes, that's correct. You can create sub-folders that way. Honestly, SAF is a pretty big can of worms. I might start a separate package just for SAF support

**Comment by gakshat14** [comment]
@crutchcorn so were you able to read files from the download directory? Because as far as I know we can't read the files from download directory.

**Comment by crutchcorn** [comment]
It seems to work for me. This is the code I'm using to do the write:

https://github.com/oceanbit/GitShark/blob/upgrade-deps/src/components/error-prompt/error-prompt-common.tsx#L55
